envoy/ssl/ssl_socket_extended_info.h

```diff
@@ -24,6 +49,31 @@ class SslExtendedSocketInfo {
    * @return ClientValidationStatus The peer certificate validation status.
    **/
   virtual ClientValidationStatus certificateValidationStatus() const PURE;
+
+  /**
+   * Only called when doing asynchronous cert validation.
+   * @param current_tls_alert the value to be latched for return after the async validation.
+   * @return ValidateResultCallbackPtr a callback used to return the validation result.
+   */
+  virtual ValidateResultCallbackPtr createValidateResultCallback(uint8_t* current_tls_alert) PURE;
```

Comment:
It is passed in to retain the current tls alert value in the callback object, so that it can be used for return in an async validation if not overridden by the validator.